### PR TITLE
Makes cover work at a distance

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Depending on aiming, a bug allows people behind cover to be hit without the cover properly activating. This fixes it, as per Presidente's request.

## Why It's Good For The Game

Ask Presidente.

## Changelog
:cl:
fix: cover triggers properly
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
What happened to the cover buff? why make cover scramble aimed zones again and fuck over anyone tryng to use the valkrie or any sniper / laser in general.

**Comment by Humonitarian** [comment]
> What happened to the cover buff? why make cover scramble aimed zones again and fuck over anyone tryng to use the valkrie or any sniper / laser in general.

That pr was meant to be a joke

**Comment by ValoTheValo** [comment]
Salt PR

**Comment by Iglpigl** [comment]
Salt PR.

**Comment by Humonitarian** [comment]
> Salt PR
#7045 implements a bugfix
#7186 reverts the bugfix
#7217 reimplements bugfix by reverting the revert to the bugfix

Salt to the salt throne

**Comment by TheShown911** [comment]
What if you fix both bugs

**Comment by Iglpigl** [comment]
> What if you fix both bugs

^^^

**Comment by ValoTheValo** [comment]
> > Salt PR
> > #7045 implements a bugfix
> > #7186 reverts the bugfix
> > #7217 reimplements bugfix by reverting the revert to the bugfix
> 
> Salt to the salt throne

Well you see shitcoder. The bug is a feature because cover is a bad mechanic. And makes the bad mechanic slightly more tolerable

**Comment by Humonitarian** [comment]
> What if you fix both bugs

Tell that to seethe pull cope request

**Comment by thedarkcrusader** [comment]
> > > Salt PR
> > > #7045 implements a bugfix
> > > #7186 reverts the bugfix
> > > #7217 reimplements bugfix by reverting the revert to the bugfix
> > 
> > 
> > Salt to the salt throne
> 
> Well you see shitcoder. The bug is a feature because cover is a bad mechanic. And makes the bad mechanic slightly more tolerable

SPCR'S ""Bugfix""  just removed  the abllity for cover to be used while standing up defeating the entire purpose of cover

**Comment by TheShown911** [comment]
I'd rather have no cover at all than broken cover.

File: code/game/machinery/deployable.dm [hunk 1]
```diff
@@ -164,13 +164,11 @@ for reference:
...
 	if (get_dist(P.starting, loc) <= 1) //Cover won't help you if people are THIS close
-		return TRUE
-	if(get_dist(loc, P.trajectory.target) > 1 ) // Target turf must be adjacent for it to count as cover
-		return TRUE
+		return 1
 	var/valid = FALSE
 	var/distance = get_dist(P.last_interact,loc)
 	if(!P.def_zone)
-		return TRUE // Emitters, or anything with no targeted bodypart will always bypass the cover
+		return 1 // Emitters, or anything with no targeted bodypart will always bypass the cover
```
Comment (SirRichardFrancis) [inline code/game/machinery/deployable.dm:171]: Bruh.

File: code/modules/tables/interactions.dm [hunk 2]
```diff
@@ -22,13 +22,11 @@
 //checks if projectile 'P' from turf 'from' can hit whatever is behind the table. Returns 1 if it can, 0 if bullet stops.
 /obj/structure/table/proc/check_cover(obj/item/projectile/P, turf/from)
 	if (get_dist(P.starting, loc) <= 1) //Tables won't help you if people are THIS close
-		return TRUE
-	if(get_dist(loc, P.trajectory.target) > 1 ) // Target turf must be adjacent for it to count as cover
-		return TRUE
+		return 1
```
Comment (SirRichardFrancis) [inline code/modules/tables/interactions.dm:25]: Bruh.

File: code/modules/tables/interactions.dm [hunk 3]
```diff
@@ -22,13 +22,11 @@
...
 	var/valid = FALSE
 	var/distance = get_dist(P.last_interact,loc)
 	if(!P.def_zone)
-		return TRUE // Emitters, or anything with no targeted bodypart will always bypass the cover
+		return 1 // Emitters, or anything with no targeted bodypart will always bypass the cover
```
Comment (SirRichardFrancis) [inline code/modules/tables/interactions.dm:29]: Bruh.

**Review by SirRichardFrancis — COMMENTED** [review]
Bruh.

**Comment by Humonitarian** [comment]
GitHub acting funny

**Pull request closed without merging**